src/System.Text.Json/tests/JsonBooleanTests.cs

```diff
@@ -0,0 +1,136 @@
+            Assert.False(jsonBoolean.Value);
+        }
+
+        [Fact]
+        public static void TestEquals()
+        {
+            var jsonBooleanTrue = new JsonBoolean(true);
+            var jsonBooleanFalse = new JsonBoolean(false);
+
+            Assert.True(jsonBooleanTrue.Equals(new JsonBoolean(true)));
+            Assert.True(new JsonBoolean(true).Equals(jsonBooleanTrue));
+
+            Assert.True(jsonBooleanFalse.Equals(new JsonBoolean(false)));
+            Assert.True(new JsonBoolean(false).Equals(jsonBooleanFalse));
+            Assert.True(new JsonBoolean().Equals(jsonBooleanFalse));
+            Assert.False(new JsonBoolean().Equals(jsonBooleanTrue));
+
+            Assert.True(jsonBooleanTrue == new JsonBoolean(true));
+            Assert.True(jsonBooleanTrue != jsonBooleanFalse);
+
+            JsonNode jsonNodeTrue = new JsonBoolean(true);
+            Assert.True(jsonBooleanTrue.Equals(jsonNodeTrue));
+
+            JsonNode jsonNodeFalse= new JsonBoolean(false);
+            Assert.True(jsonBooleanFalse.Equals(jsonNodeFalse));
+
+            IEquatable<JsonBoolean> jsonBooleanIEquatableTrue = jsonBooleanTrue;
+            Assert.True(jsonBooleanIEquatableTrue.Equals(jsonBooleanTrue));
+            Assert.True(jsonBooleanTrue.Equals(jsonBooleanIEquatableTrue));
+
+            IEquatable<JsonBoolean> jsonBooleanIEquatableFalse = jsonBooleanFalse;
+            Assert.True(jsonBooleanIEquatableFalse.Equals(jsonBooleanFalse));
+            Assert.True(jsonBooleanFalse.Equals(jsonBooleanIEquatableFalse));
+
+            Assert.False(jsonBooleanTrue.Equals(null));
+
+            object jsonNumberCopyTrue = jsonBooleanTrue;
+            object jsonNumberObjectTrue = new JsonBoolean(true);
+            Assert.True(jsonNumberCopyTrue.Equals(jsonNumberObjectTrue));
+            Assert.True(jsonNumberCopyTrue.Equals(jsonNumberObjectTrue));
+            Assert.True(jsonNumberObjectTrue.Equals(jsonBooleanTrue));
+
+            object jsonNumberCopyFalse = jsonBooleanFalse;
+            object jsonNumberObjectFalse = new JsonBoolean(false);
+            Assert.True(jsonNumberCopyFalse.Equals(jsonNumberObjectFalse));
+            Assert.True(jsonNumberCopyFalse.Equals(jsonBooleanFalse));
+            Assert.True(jsonNumberObjectFalse.Equals(jsonBooleanFalse));
+
+            Assert.False(jsonBooleanTrue.Equals(new Exception()));
+
+            JsonBoolean jsonBooleanNull = null;
+            Assert.False(jsonBooleanTrue == jsonBooleanNull);
+            Assert.False(jsonBooleanNull == jsonBooleanTrue);
+
+            Assert.True(jsonBooleanTrue != jsonBooleanNull);
+            Assert.True(jsonBooleanNull != jsonBooleanTrue);
+        }
+
+        [Fact]
+        public static void TestGetHashCode()
+        {
+            var jsonBooleanTrue = new JsonBoolean(true);
+            var jsonBooleanFalse = new JsonBoolean(false);
+
+            Assert.Equal(jsonBooleanTrue.GetHashCode(), new JsonBoolean(true).GetHashCode());
```

Comment:
nit: Also add a test actually verifying the hashcode value itself (and not just equality).
```C#
Assert.Equal(<whatever expected int>, jsonBooleanTrue.GetHashCode());
Assert.Equal(<whatever expected int>, jsonBooleanFalse.GetHashCode());
```